docs/spec.md

```diff
@@ -446,18 +435,19 @@ to that of the derivation (or derivations).
 
 
 
-## Traversable
-
-#### Dependencies
-
-  1. Functor
-  1. Foldable
+### Traversable
 
-#### Methods
+In the following signature `Applicative<U>` means that a value must
+not only match `Applicative` signature, but also fully support `Applicative` algebra.
 
-  1. `traverse :: (Traversable t, Applicative f) => Type t ~> (Type f, (a → f b), t a) → f (t b)`
+```js
+Traversable<T> {
+  traverse: (Applicative<U>, a => U<b>, T<a>) => U<T<b>>
+}
+```
```

Comment:
I think we're getting confused here (and thus my idea might be a bad one):

- Module type: You think `Applicative<U>`, I think `U where U :> Applicative`
- Any instance: You think `U`, I think `U<*>` or `U<any>`
- Typed instance: You think `U<a>`, I think `U<a>`

So I think part of the confusion is because we see two completely different things.

To help you see where I'm coming from, here's a Scala equivalent for the same interface:

```scala
trait Traversable[T[_]] extends Functor[T[_]] with Foldable[T[_]] {
  def traverse[U[_] <: Applicative[U[_], A, B](
    type: U[_], func: A => U[B], initial: T[A]
  ): U[T[B]]
}
```

(of course, idiomatic Scala would use the FL variant)